### ISSUE TITLE
`indentation` never reaches a line break another rule writes in the same run

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#353]
Stylelint runs the rules of a plugin in the order the configuration spells them, and it runs each of them once. A line break written by one rule is therefore invisible to `indentation` whenever `indentation` is listed first, and the line that break begins is left without its indent.

## The run

Fixture `a {\n\tcolor: red; top: 0;\n}\n`, both rules at once, only the order changed:

```
declaration-block-semicolon-newline-after first -> "a {\n  color: red;\n  top: 0;\n}\n"   nothing left to report
indentation first                               -> "a {\n  color: red;\n top: 0;\n}\n"    one warning left
```

The second declaration is carried onto a line of its own, and the line keeps the single space that stood behind the semicolon instead of the indent the configuration asks for. A second run of `--fix` repairs it, and the first run reports nothing at all.

## Why

`lintPostcssResult.mjs` of `stylelint@17.14.1` sorts the rules of a run by their place in Stylelint's own registry, and a plugin's rules are not in it:

```js
const rulesOrder = new Map(Object.keys(rules).map((ruleName, index) => [ruleName, index]))

ruleNames = Object.keys(config.rules).sort((a, b) => (rulesOrder.get(a) ?? -1) - (rulesOrder.get(b) ?? -1))
```

Every rule of this plugin resolves to `-1`, the sort is stable, and so the order is the one the user's configuration happens to spell. `lintPostcssResult` is called once per lint; there is no loop to a fixed point.

## The counterparts

Three rules were measured to answer differently under the two orders:

- `block-closing-brace-newline-before`
- `block-opening-brace-newline-after`
- `declaration-block-semicolon-newline-after`

The corpus reached these three; the class is every rule that writes a line break, since each of them begins a line `indentation` would have measured.

## What ends it

A rule that writes a line break has to write the indent behind it, rather than leave a line for `indentation` to find. This one is heavier than the two sibling classes: it is not a setting to read but a calculation to share, so the indent a rule writes and the indent `indentation` measures come from the same place.

The measurement ran 79 rules under 208 primary options over eight fixtures of plain CSS.

One of the four classes of [#356](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/356), which is about the class rather than about any member of it.
